### PR TITLE
Register HubiC account without user intervention using URL for credentials

### PR DESCRIPTION
**Pull request by sdelrio**
To register HubiC account without user intervention

Using: https://yourserver.com/autoregister/?client=hubic&password=mypassword&hubic_user=myhubiclogin&hubic_password=myhubicpassword

Added fields to config.php.sample:

'autoredirect_uri'. I set by default to localhost/callback since it will be the host with the webserver running doing the request.

### CODE REVIEW

File: www/simple.php [hunk 1]
```diff
@@ -411,6 +547,9 @@ function getScheme() {
 	header('HTTP/1.0 204 No Content');
 //	header('HTTP/1.0 200 OK'); // dulwich swift-repo does not understand 204
 	nocache();
+} else if($mode=='autoregister') {
```
Comment (oderwat) [inline www/simple.php]: I already merged and tested your PR .. but I think there should be an "autoregister sucessfull"  message in this case.

**Comment by oderwat** [comment]
I changed the code such that a successful autoregister also lands on the success page. I hope that this does not break something for your friend (as it will do a redirection first instead of just returning 200 OK)

**Comment by sdelrio** [comment]
It's OK, I will take care to check the last url on /success instead when checking with the other part for success.

**Comment by oderwat** [comment]
I also changed the docs and removed the need for the callback url in the config.php.. could you please test if that works for you? https://github.com/oderwat/hubic2swiftgate/pull/33

**Comment by sdelrio** [comment]
For me doesn't works because the initial petition could be http://<IP1>:9080, but inside the container is running apache on :80, and that way the callback will forward me to localhost:9080, that is not watching.

But, pull that change and I can make a condition that looks if exist that variable
```php
if (in_array ($clients[$client]['autoredirect_uri'])) { ...
```
then use that one, else use your URL

```php
urlencode(getScheme()."://".$_SERVER['SERVER_NAME'].$port.$_prefix."/callback/"),
```

**Comment by oderwat** [comment]
Yes that was my thinking for the case that one needs to change it. Would be nice if you add that. Please add that detail to the documentation too.

**Comment by sdelrio** [comment]
I'm on it.

You already have this on line 110:

```php
$redirect_uri=getScheme()."://".$_SERVER['SERVER_NAME'].$port.$_prefix."/callback/";
```
So, instead of writing:

```php
urlencode(getScheme()."://".$_SERVER['SERVER_NAME'].$port.$_prefix."/callback/"),
```
I will just use the variable `$redirect_uri`.

**Pull request merged**